### PR TITLE
Allow images to exist in several versions

### PR DESCRIPTION
**Pull request by sampaccoud**
## Purpose

We need to build 2 different Docker images for Elasticsearch (0.9 and 6.2.4). The name of the images is clashing because it was not planned. 

## Proposal

This commit introduces nested directories with first the name of the image, then the tag of the image.

### CODE REVIEW

**Comment by sampaccoud** [comment]
@jmaupetit something weird with this code that I don't understand. Probably some trick in bash that I don't know.
 
works:

    $ bin/publish elasticsearch:6.2.4

works:

    $ bin/publish elasticsearch:0.9

works for the first image and fails on the second

    $ bin/publish elasticsearch
      image path does not exist: docker/images/elasticsearch/6.2.4/Dockerfile

File: .circleci/config.yml [hunk 1]
```diff
@@ -51,13 +51,12 @@ jobs:
           command: echo "$DOCKER_PASS" | docker login -u "$DOCKER_USER" --password-stdin
 
       # We only publish build images upon new git tags with the following
-      # pattern: `service-version`, e.g. `nginx-1.13`
+      # pattern: <image name>[-<image tag>], e.g. nginx or nginx-1.13
```
Comment (jmaupetit) [inline .circleci/config.yml]: The pattern should be : `<image name>[:<image tag>], e.g. nginx or nginx:1.13`

File: .circleci/config.yml [hunk 2]
```diff
@@ -100,16 +99,16 @@ workflows:
             tags:
               only: /.*/
 
-      # We publish a new service release for each tag respecting the following
-      # pattern: <service>-<version>, e.g. nginx-1.13
+      # We publish new images for each git tag respecting the following
+      # pattern: <image name>[-<image tag>], e.g. nginx or nginx-1.13
```
Comment (jmaupetit) [inline .circleci/config.yml]: Same thing here.

**Review by jmaupetit — APPROVED** [review]
Great improvement, thanks!

**Pull request merged**